structure.go

```diff
@@ -123,6 +123,7 @@ type MediaPlaylist struct {
 	Map            *Map // EXT-X-MAP is optional tag specifies how to obtain the Media Initialization Section (default map for the playlist)
 	WV             *WV  // Widevine related tags outside of M3U8 specs
 	lock           *sync.Mutex
+	dirty          bool
```

Comment:
please add a comment indicating what elements this lock is protecting

Reply:
fixed